source/geod24/LocalRest.d

```diff
@@ -497,8 +499,16 @@ public final class RemoteAPI (API) : API
                                 await_msgs ~= Variant(cmd);
                         });
 
-                    // now handle any leftover messages after any sleep() call
-                    if (!isSleeping())
+                    // if this node was sleeping and then terminated then we cannot process any leftover requests from
+                    // other nodes, however we must notify other nodes that their requests have failed.
```

Comment:
Why do we need to notify of failure ? This is the difference between a "clean" and an immediate shutdown. In a clean shut down you get an RST (or the equivalent for your protocol), but in an immediate shutdown you just time out.
I would think it is more robust to always expose immediate shutdown semantics, no ?

Reply:
Right. I was thinking in terms of what is more useful to the programmer, as a debugging tool. But maybe having immediate shutdown semantics makes much more sense.

In that case, the solution is much simpler. We just throw the preallocated exception in the event loop.